src/tests/conformance/post_handshake/validators.rs

```diff
@@ -55,3 +71,259 @@ async fn c015_TM_VALIDATOR_LIST_COLLECTION_node_should_send_validator_list() {
     };
     perform_expected_message_test(Default::default(), &check).await;
 }
+
+fn create_sha512_half_digest(buffer: &Vec<u8>) -> [u8; 32] {
+    let mut hasher = Sha512::new();
+    hasher.update(buffer);
+    let result = hasher.finalize();
+
+    // we return 32 bytes of 64-byte result
+    let mut signature = [0u8; 32];
+    for i in 0..32 {
```

Comment:
Instead of the loop it could be possible to do this:
`signature.copy_from_slice(&result[..32]);`